src/server/shared/Utilities/Util.cpp

```diff
@@ -509,32 +509,21 @@ bool Utf8FitTo(const std::string& str, std::wstring const& search)
     return true;
 }
 
-void utf8printf(FILE* out, const char *str, ...)
-{
-    va_list ap;
-    va_start(ap, str);
-    vutf8printf(out, str, &ap);
-    va_end(ap);
-}
-
-void vutf8printf(FILE* out, const char *str, va_list* ap)
+void vutf8printf(FILE* out, std::string const& str)
 {
 #if PLATFORM == PLATFORM_WINDOWS
-    char temp_buf[32*1024];
-    wchar_t wtemp_buf[32*1024];
-
-    size_t temp_len = vsnprintf(temp_buf, 32*1024, str, *ap);
-    //vsnprintf returns -1 if the buffer is too small
-    if (temp_len == size_t(-1))
-        temp_len = 32*1024-1;
+    std::wstring wstr;
+    Utf8toWStr(str, wstr);
 
-    size_t wtemp_len = 32*1024-1;
-    Utf8toWStr(temp_buf, temp_len, wtemp_buf, wtemp_len);
+    //TODO Improve me
+    char temp_buf[32 * 1024];
+    CharToOemBuffW(&wstr.c_str()[0], &temp_buf[0], wstr.length());
```

Comment:
This will lead to random characters in front of the string.
Anyone got an idea?